### DISCUSSION TITLE
Stopwatch example

### DISCUSSION THREAD

**Discussion by Gcopper1984** [pybricks/support#507] [Show and tell]
Hello,

    I did not found any example of StopWatch class in documentation so here it is a simple example:

```python
from pybricks.tools import StopWatch

stopwatch = StopWatch()

clockTime = 1000

while True:
    time = stopwatch.time()
    if time >= clockTime:
        print("Tick: ", time)
        stopwatch.reset()

```

I saw that it has a millisecond precision, is it correct?

Thank you

**Comment by dlech** [comment]
Yes, it is milliseconds.

**Reply by Gcopper1984** [reply]
Hi, how much is the maximun timespan measured by stopwatch until it overflow?

Thank you.

**Reply by dlech** [reply]
It uses a 32-bit integer, so approx 4 billion milliseconds.

**Comment by Gcopper1984** [comment]
It's more or less 46 days, right? And when It overflows It start from 0 or
launch some errore/exception?

Thank you

Il sab 23 ott 2021, 21:05 David Lechner ***@***.***> ha
scritto:

**Reply by dlech** [reply]
it will just go back to 0
